src/Propel/Generator/Builder/Om/QueryBuilder.php

```diff
@@ -627,10 +635,9 @@ public function findPk(\$key, ConnectionInterface \$con = null)
             return null;
         }";
         if ($table->hasCompositePrimaryKey()) {
-            $pks = [];
-            foreach ($table->getPrimaryKey() as $index => $column) {
-                $pks[] = "\$key[$index]";
-            }
+            $numberOfPks = count($table->getPrimaryKey());
+            $pkIndexes = range(0, $numberOfPks - 1);
+            $pks = preg_filter('/(\d+)/', '$key[${1}]', $pkIndexes); // put ids into "$key[]"
```

Comment:
Well, this is pretty cool :)
I've compared it with regular foreach loop and it's really faster

Reply:
Nice! I read on stackoverflow, that it is faster, but was too lazy to test (I changed it because I find it more readable).
And now I know it really is faster. Thanks!